### PR TITLE
Rework benchmarking

### PR DESCRIPTION
**Pull request by FabioLuporini**

### CODE REVIEW

File: benchmarks/user/README.md [hunk 1]
```diff
@@ -0,0 +1,233 @@
+# Benchmarking a Devito Operator
+
+## Running benchmarks
+
+`benchmark.py` implements a minimalist framework to evaluate the performance of
+a Devito Operator while varying
+
+* the problem size (e.g., shape of the computational grid)
```
Comment (georgebisbas) [inline benchmarks/user/README.md]: Trivial :
Maybe size and shape/ size of the shape...
* the problem size (e.g., shape size of the computational grid)
I know it is defined as `shape` but sounds like weird IMHO....

File: benchmarks/user/README.md [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+* the discretization (e.g., space- and time-order of the input/output fields),
+* the simulation time,
```
Comment (georgebisbas) [inline benchmarks/user/README.md]: e.g. number of timesteps?

File: benchmarks/user/README.md [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+## A step back: configuring your machine for reliable benchmarking
+
+If you are tempted to use your laptop to run a benchmark, you may want to
+reconsider: heat and power management may affect the results you get in an
+unpredictable way.
+
+It is important that *both* the Python process running Devito (process*es* if
+running with MPI) and the OpenMP threads spawned while running an Operator are
+pinned to specific CPU cores, to get reliable and determinisic results. There
+are several ways to achieve this.
```
Comment (georgebisbas) [inline benchmarks/user/README.md]: ways to achieve this **:**

File: benchmarks/user/README.md (hunk 2 above)
Comment (FabioLuporini) [reply benchmarks/user/README.md]: in fact no, only the simulation time in milliseconds con be provided, from which the number of timesteps will be derived

Comment (mloubout) [reply benchmarks/user/README.md]: In millisecond

File: benchmarks/user/README.md [hunk 4]
```diff
@@ -0,0 +1,233 @@
+# Benchmarking a Devito Operator
+
+## Running benchmarks
+
+`benchmark.py` implements a minimalist framework to evaluate the performance of
+a Devito Operator while varying
+
+* the problem size (e.g., shape of the computational grid)
+* the discretization (e.g., space- and time-order of the input/output fields),
```
Comment (mloubout) [inline benchmarks/user/README.md]: so and to of wavefields

File: benchmarks/user/README.md [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+optimizations applied by an Operator, how auto-tuning is getting along, and to
+emit more performance metrics.
+
+You can also set `DEVITO_DEBUG_COMPILER=1` to emit the command used to compile
+the generated code.
+
+## Tips
+
+* The 'O3' mode has been found to be particularly beneficial in TTI Operators,
+  by resulting in significant reductions in operation count.
```
Comment (mloubout) [inline benchmarks/user/README.md]: With intel arch

File: benchmarks/user/README.md [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+htop
+```
+or to keep the memory footprint under control:
+```
+watch numastat -m
+```
+
+## Known limitations and possible work arounds
+
+ * The DSE `aggressive` mode will not work with backend compilers that are not
```
Comment (mloubout) [inline benchmarks/user/README.md]: It will work just won't be efficient

File: benchmarks/user/README.md (hunk 1 above)
Comment (FabioLuporini) [reply benchmarks/user/README.md]: mmhh, to be fair, I think it's quite clear the way it is

File: benchmarks/user/README.md (hunk 2 above)
Comment (FabioLuporini) [reply benchmarks/user/README.md]: right, fixing

File: benchmarks/user/README.md (hunk 3 above)
Comment (FabioLuporini) [reply benchmarks/user/README.md]: right, fixing

File: benchmarks/user/README.md (hunk 4 above)
Comment (FabioLuporini) [reply benchmarks/user/README.md]: I think I prefer "input/output fields" because one day we may want to use this for something that is not waves

**Comment by FabioLuporini** [comment]
@mloubout @georgebisbas addressed some of your comments

**Comment by FabioLuporini** [comment]
merged.

**Pull request merged**